src/cloudWatchLogs/changeLogSearch.ts

```diff
@@ -18,28 +16,27 @@ import { isViewAllEvents, TimeFilterResponse, TimeFilterSubmenu } from './timeFi
  * @param oldUri
  * @returns Undefined if cancelled and the newData otherwise.
  */
+
 export async function getNewData(
-    registry: LogStreamRegistry,
     param: 'filterPattern' | 'timeFilter',
-    oldUri: vscode.Uri
+    oldData: CloudWatchLogsData
 ): Promise<CloudWatchLogsData | undefined> {
-    const oldData = registry.getLogData(oldUri)
-    if (!oldData) {
-        throw new Error(`cwl: LogStreamRegistry did not contain ${String(oldUri)}`)
-    }
+    // We must deepcopy the parameters so that we don't change their original value in oldData
     const newData: CloudWatchLogsData = {
         ...oldData,
+        parameters: { ...oldData.parameters },
```

Comment:
Oops I forgot to include `{ }` in my suggestion to copy objects. But you were able to figure it out.